go/websocketbridge/websocket_bridge_test.go

```diff
@@ -0,0 +1,233 @@
+	exocom := exocomMock.New()
+	go func() {
+		err := exocom.Listen(port)
+		if err != nil && err != http.ErrServerClosed {
+			log.Fatal(err)
+		}
+	}()
+	return exocom
+}
+
+// Cucumber step definitions
+// nolint gocyclo
+func FeatureContext(s *godog.Suite) {
+	var exocom *exocomMock.ExoComMock
+	var websocketBridgeInstance *websocketbridge.WebsocketBridge
+	var clientWebsocket *websocket.Conn
+	var clientCount int
+	exoPort := 4100
+	clientPort := 5435
+	var outgoingMessageId string
+	var savedError error
+
+	s.BeforeSuite(func() {
+		exocom = newExocom(exoPort)
+	})
+
+	s.BeforeScenario(func(interface{}) {
+		outgoingMessageId = ""
+		savedError = nil
+		clientWebsocket = nil
+		clientCount = 0
+	})
+
+	s.AfterScenario(func(interface{}, error) {
+		exocom.Reset()
+		err := websocketBridgeInstance.Close()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.AfterSuite(func() {
+		err := exocom.Close()
+		if err != nil {
+			panic(err)
+		}
+	})
+
+	s.Step(`a websocket bridge connected to Exocom`, func() error {
+		websocketBridgeInstance = &websocketbridge.WebsocketBridge{}
+		config := exorelay.Config{Host: "localhost", Port: exoPort, Role: "websocket-test"}
+		err := websocketBridgeInstance.Open(config, clientPort)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
+	s.Step(`receiving this message from the client:`, func(payloadStr *gherkin.DocString) error {
+		origin := "http://localhost/"
+		url := fmt.Sprintf("ws://localhost:%v/ws", clientPort)
+		var err error
+		clientWebsocket, err = websocket.Dial(url, "", origin)
+		if err != nil {
+			return err
+		}
+		var message structs.Message
+		err = json.Unmarshal([]byte(payloadStr.Content), &message)
+		if err != nil {
+			return err
+		}
+		outgoingMessageId = message.ID
+		marshaledMessage, err := json.Marshal(message)
+		_, err = clientWebsocket.Write(marshaledMessage)
+		return err
+	})
+
+	s.Step(`^receiving a message "([^"]*)" from (\d+) different clients$`, func(messageName string, count int) error {
+		clientCount = count
+		var err error
+		url := fmt.Sprintf("ws://localhost:%v/ws", clientPort)
+		for i := 0; i < clientCount; i++ {
+			clientWebsocket, err = websocket.Dial(url, "", "http://localhost:4000")
+			if err != nil {
+				return err
+			}
+			message := structs.Message{Name: messageName, ID: uuid.NewV4().String()}
+			marshaledMessage, err := json.Marshal(message)
+			if err != nil {
+				return err
+			}
+			_, err = clientWebsocket.Write(marshaledMessage)
+			if err != nil {
+				return err
+			}
+		}
+		return err
+	})
+
+	s.Step(`receiving this message from exocom:`, func(payloadStr *gherkin.DocString) error {
+		var message structs.Message
+		err := json.Unmarshal([]byte(payloadStr.Content), &message)
+		if err != nil {
+			return err
+		}
+		err = exocom.WaitForConnection()
+		if err != nil {
+			return err
+		}
+		if message.SessionID != "" {
+			for key := range websocketBridgeInstance.SocketMap {
+				message.SessionID = key
+			}
+		}
```

Comment:
I'm not a fan of what this is doing. Can we set the session id based on a message sent to exocom? I'd prefer to keep the socket map private